ravendb/rvn_server.go

```diff
@@ -0,0 +1,824 @@
+			time.Sleep(time.Second * 2)
+		} else if err == nil {
+			break
+		} else {
+			return ns, errors.New("Unable to SSH to " + hostAndPort + " because " + err.Error())
+		}
+	}
+	defer conn.Close()
+
+	files, err := listFiles(conn, "/etc/ravendb")
+	if err != nil {
+		return ns, err
+	}
+	ns.Assets = make(map[string][]byte)
+	for _, file := range files {
+		_, fileName := filepath.Split(file)
+
+		contents, err := readFileContents(file, stdoutBuf, conn)
+		if err != nil {
+			return ns, err
+		}
+		if contents == nil {
+			contents = make([]byte, 0) // empty file
+		}
+		ns.Assets[fileName] = contents
+	}
+
+	ns.Settings = make(map[string]interface{})
+	if file, ok := ns.Assets["settings.json"]; ok {
+		err = json.Unmarshal(file, &ns.Settings)
+		if err != nil {
+			stdoutBuf.WriteString("Failed to parse settings.json\n")
+			stdoutBuf.Write(file)
+			return ns, err
+		}
+		delete(ns.Assets, "settings.json")
+	}
+	//workaround to convert unmarshalled map[string]interface{} values to string.
+	for key := range ns.Settings {
+		ns.Settings[key] = fmt.Sprintf("%v", ns.Settings[key])
+	}
+
+	if license, ok := ns.Assets["license.json"]; ok {
+		ns.Licence = license
+		delete(ns.Assets, "license.json")
+	}
+
+	if cert, ok := ns.Assets["certificate.pfx"]; ok {
+		ns.ClusterCertificate = cert
+		delete(ns.Assets, "certificate.pfx")
+	}
+
+	store, err := getStore(sc, index)
+	if err != nil {
+		return ns, err
+	}
+	buildNumber := operations.OperationGetBuildNumber{}
+	err = executeWithRetries(store, &buildNumber)
+	if err != nil {
+		return ns, err
+	}
+
+	ns.Version = strconv.Itoa(buildNumber.BuildVersion)
+
+	ns.Host = publicIP
+	ns.TcpUrl = ns.Settings["PublicServerUrl"].(string)
+	ns.HttpUrl = ns.Settings["PublicServerUrl.Tcp"].(string)
+	if unsecuredAccessAllowed, ok := ns.Settings["Security.UnsecuredAccessAllowed"]; ok {
+		ns.Insecure = unsecuredAccessAllowed == "PublicNetwork"
+	}
+
+	delete(ns.Settings, "PublicServerUrl")
+	delete(ns.Settings, "PublicServerUrl.Tcp")
+	delete(ns.Settings, "Security.UnsecuredAccessAllowed")
+
+	return ns, nil
+}
+
+func (sc *ServerConfig) deployServer(publicIP string, index int) (err error) {
+	var stdoutBuf bytes.Buffer
+	var conn *ssh.Client
+	defer func() {
+		log.Println(stdoutBuf.String())
+	}()
+	ravenPackageUrl := "https://daily-builds.s3.us-east-1.amazonaws.com/ravendb_" + sc.Package.Version + sc.Package.Arch
+
+	signer, err := ssh.ParsePrivateKey(sc.SSH.Pem)
+	if err != nil {
+		return err
+	}
+	authConfig := &ssh.ClientConfig{
+		User:            sc.SSH.User,
+		Auth:            []ssh.AuthMethod{ssh.PublicKeys(signer)},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         1 * time.Minute,
+	}
+	hostAndPort := net.JoinHostPort(publicIP, fmt.Sprint(sc.SSH.getPort()))
+	for i := 0; i <= numberOfIterationsExecuteWithRetries; i++ {
+		conn, err = ssh.Dial("tcp", hostAndPort, authConfig)
+		if err != nil && i < numberOfIterationsExecuteWithRetries {
+			time.Sleep(time.Second * 2)
```

Comment:
Log here? Also, isn't that the same as the previous one? Can we avoid duplication?